### PR TITLE
OP-16464: integrate pipeline trigger to verify fiat permissions

### PR DESCRIPTION
**Pull request by sudhakaropsmx**
https://devopsmx.atlassian.net/browse/OP-16464

Pipeline RBAC implementation commits have ORCA, Front50, and Fiat, Gate is missing implementation.
As part of the Pipeline RBAC implementation these changes required before trigger pipeline to verify the permissions against with fiat

https://github.com/OpsMx/gate/pull/279

### CODE REVIEW

File: gate-web/src/main/groovy/com/netflix/spinnaker/gate/controllers/PipelineController.groovy [hunk 1]
```diff
@@ -272,7 +272,7 @@ class PipelineController {
   }
 
   @Operation(summary = "Trigger a pipeline execution")
-  @PreAuthorize("hasPermission(#application, 'APPLICATION', 'EXECUTE')")
+  @PreAuthorize("hasPermission(#application, 'APPLICATION', 'EXECUTE') && hasPermission(#pipelineNameOrId, 'PIPELINE', 'EXECUTE')")
```
Comment (aman-agrawal) [inline gate-web/src/main/groovy/com/netflix/spinnaker/gate/controllers/PipelineController.groovy:275]: can u pls explain why we made this change ?

Comment (sudhakaropsmx) [reply gate-web/src/main/groovy/com/netflix/spinnaker/gate/controllers/PipelineController.groovy:275]: If we enable the Pipeline RBAC before trigger pipeline to verify the permissions against with fiat

Comment (aman-agrawal) [reply gate-web/src/main/groovy/com/netflix/spinnaker/gate/controllers/PipelineController.groovy:275]: 1. oss dont hav pipeline level rbac, only oes hav ?
2. If yes, then why only selected controller, not other controllers as well?
3. If someone gave execute access for some pipeline but not to application, is this scenario possible?

Comment (sudhakaropsmx) [reply gate-web/src/main/groovy/com/netflix/spinnaker/gate/controllers/PipelineController.groovy:275]: oes will have pipeline rbac.
Pipeline trigger will from gate to echo and to orca. remain verification have in orca and front50.
first priority Application level and then Pipeline.
This feature is requested by customer WU and currently using this feature.

Comment (aman-agrawal) [reply gate-web/src/main/groovy/com/netflix/spinnaker/gate/controllers/PipelineController.groovy:275]: in a call, @sudhakaropsmx mentioned that other checks r implemented in other services like orca.
still, i hav one doubt, for few controllers we r allowing access from gate to orca & checking in orca for rbac.
but for some controllers, we r checking rbac in gate itself, any particular reason ?
-> but as this is not a blocker, approving the PR.

**Review by aman-agrawal — APPROVED** [review]

**Pull request merged**